lib/chef/provider/package/chocolatey.rb

```diff
@@ -282,6 +283,25 @@ def parse_list_output(*args)
         def lowercase_names(names)
           names.map(&:downcase)
         end
+
+        def find_list_options
+          username = ""
+          password = ""
+          options = new_resource.options.split(" ")
+          if (options & ["-u", "--user"]).any?
+            username = options.include?("-u") ? find_credential("-u", options) : find_credential("--user", options)
+          end
+          if (options & ["-p", "--password"]).any?
+            password = options.include?("-p") ? find_credential("-p", options) : find_credential("--password", options)
+          end
+          [username, password].compact.join(" ").strip
+        end
+
+        def find_credential(element, array)
+          index = array.find_index(element) + 1
+          next_element = array.at(index)
+          next_element ? element + " " + next_element : nil
+        end
```

Comment:
This now feels like we're going in the wrong direction trying to solve this problem.

We probably need to back up and add `new_resource.list_options` as something that users can set which is separate to `new_resource.options`.  For common things like user and password those should go onto additional `new_resource.user` and `new_resource.password` fields.

I could potentially be persuaded to avoid the duplication between `list_options` and `options` but then `options` should get passed to both, and any options which cannot apply to both then must be made into properties on the new_resource.

What I don't know is what the overlap looks like between the different options that those CLI commands support.  If the union is large, then users probably want `options` passed to both, if the union is small then users probably don't.  Based on the code here it looks like the union is small so that users probably want `options` to only go to the main install or remove command and that `new_resource.list_options` should be a separate.

Either way though we should just use properties instead of doing this parsing out of the options argument.  The creds are important enough they should be top-level properties.